pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+from collections import OrderedDict
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        # We skip summary, deprecation warning, and extended
+        # summary
+        "Parameters",
+        "Attributes",
+        "Methods",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "Raises",
+        "Warns",
+        "Warnings",
+        "See Also",
+        "Notes",
+        "References",
+        "Examples",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Sort the numpydoc sections into the numpydoc order."""
+        new_sections = OrderedDict([sections.popitem(last=False)])
+        new_sections.update(
```

Comment:
Can't we remove L36 and just return a sorted `OrderedDict`?

Reply:
I don't see an `OrderedDict.sort` method [here](https://docs.python.org/3/library/collections.html#collections.OrderedDict).  and the implementation [here](https://github.com/python/cpython/blob/8b34e914bba2ccd6ae39609410db49d0beb19cb1/Lib/collections/__init__.py#L78-L329) looks like the ordering is kept by a linked list rather than a list I can sort.

To merge line 36 with 37-44, I would need a consistent name for the summary/deprecation warning/extended summary section, which currently uses the summary as the name.  I can look into that, as it would simplify other things.

Reply:
Initial section now named "Summary" unless the initial section is zero lines, in which case the initial section shares a name with the next section and gets disappeared by the `dict` constructor.